src/TitleDetails/AddPieceModal/AddPieceModal.js

```diff
@@ -0,0 +1,206 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import { Field } from 'react-final-form';
+import {
+  get,
+  includes,
+} from 'lodash';
+
+import {
+  Button,
+  Checkbox,
+  Col,
+  Modal,
+  ModalFooter,
+  Row,
+  TextArea,
+  TextField,
+} from '@folio/stripes/components';
+import {
+  Pluggable,
+  stripesShape,
+  withStripes,
+} from '@folio/stripes/core';
+import stripesFinalForm from '@folio/stripes/final-form';
+import {
+  FieldDatepickerFinal,
+  FieldSelectFinal,
+  validateRequired,
+} from '@folio/stripes-acq-components';
+
+import {
+  INVENTORY_RECORDS_TYPE,
+} from '../constants';
+
+const AddPieceModal = ({
+  close,
+  createInventoryValues,
+  form,
+  handleSubmit,
+  instanceId,
+  locations,
+  onCheckIn,
+  pieceFormatOptions,
+  stripes,
+}) => {
+  const addItem = form.mutators.setItemValue;
+  const formValues = get(form.getState(), 'values', {});
+  const { format, id, itemId, locationId } = formValues;
+  const isLocationRequired = includes(createInventoryValues[format], INVENTORY_RECORDS_TYPE.instanceAndHolding);
+  const isAddItemRequired = includes(createInventoryValues[format], INVENTORY_RECORDS_TYPE.all);
+  let isAddItemButtonDisabled = true;
+
+  if (!itemId && locationId && isAddItemRequired) {
+    isAddItemButtonDisabled = false;
+  }
+
+  const disabledButtonProps = isAddItemButtonDisabled ? { disabled: isAddItemButtonDisabled } : {};
+  const labelId = id ? 'ui-receiving.piece.addPieceModal.editTitle' : 'ui-receiving.piece.addPieceModal.title';
+
+  const footer = (
+    <ModalFooter>
+      <Button
+        buttonStyle="primary"
+        data-test-add-piece-save
+        onClick={handleSubmit}
+      >
+        <FormattedMessage id="ui-receiving.piece.actions.save" />
+      </Button>
+      <Button
+        data-test-add-piece-check-in
+        onClick={() => onCheckIn(formValues)}
```

Comment:
is it possible to wrap with usecallback?

Reply:
fixed